arcgis-ios-sdk-samples/Scenes/Animate image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,211 @@
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationServiceURL)
+        let scene = AGSScene(basemap: AGSBasemap(baseLayer: worldDarkGrayBasemap))
+        // Create a surface and add it to the scene.
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        scene.baseSurface = surface
+        // Create an envelope of the pacific southwest sector for displaying the image frame.
+        let pointForImageFrame = AGSPoint(x: -120.0724273439448, y: 35.131016955536694, spatialReference: .wgs84())
+        pacificSouthwestEnvelope = AGSEnvelope(center: pointForImageFrame, width: 15.09589635986124, height: -14.3770441522488)
+        return scene
+    }
+    
+    func startAnimation(_ fps: Double) {
+        // Invalidate timer to stop previous ongoing animation.
+        frameRateTimer?.invalidate()
+        // Duration or interval between frames in seconds.
+        let duration = 1 / Double(fps)
+        let timer = Timer(timeInterval: duration, repeats: true) { [weak self] _ in
+            self?.setImageFrame()
+        }
+        self.frameRateTimer = timer
+        // Add the timer to common mode run loop, so the timer is not effected by UI events.
+        RunLoop.main.add(timer, forMode: .common)
+    }
+    
+    /// Load the URLs for the overlay images.
+    ///
+    /// - Parameter completion: If it loads successfully, enable the UI components in the completion closure.
+    func loadImageURLs(completion: () -> Void) {
+        // The images need to be added to the project with folder reference.
+        guard let pacificSouthWestURLs = Bundle.main.urls(forResourcesWithExtension: "png", subdirectory: "PacificSouthWest") else {
+            return
+        }
+        // Sort the image URLs by their relative pathnames.
+        imageURLs = pacificSouthWestURLs.sorted { $0.relativeString < $1.relativeString }
+        completion()
+    }
+    
+    /// Set the image at current index to the image overlay.
+    func setImageFrame() {
+        let imageFrame = AGSImageFrame(url: imageURLs[loopCounter.currentIndex], extent: pacificSouthwestEnvelope)
+        imageOverlay.imageFrame = imageFrame
+        loopCounter.next()
+    }
+    
+    func getOpacityString() -> String {
+        return String(format: "Opacity: %.0f%%", opacitySlider.value * 100)
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func sliderValueChanged(_ slider: UISlider) {
+        imageOverlay.opacity = slider.value
+        opacityLabel.text = getOpacityString()
+    }
+    
+    @IBAction func playPauseButtonTapped(_ button: UIBarButtonItem) {
+        if isAnimating {
+            frameRateTimer?.invalidate()
+        } else {
+            startAnimation(frameRate)
+        }
+        isAnimating.toggle()
+    }
+    
+    @IBAction func frameRateButtonTapped(_ button: UIBarButtonItem) {
```

Comment:
Should I only create once and keep a reference to the alertVC, as it won't change for this sample?

Reply:
No as the memory footprint is pretty small.